databricks/sdk/core.py

```diff
@@ -1118,3 +1120,82 @@ def _redacted_dump(self, prefix: str, body: str) -> str:
             return "\n".join([f'{prefix}{line}' for line in json.dumps(raw, indent=2).split("\n")])
         except JSONDecodeError:
             return f'{prefix}[non-JSON document of {len(body)} bytes]'
+
+
+class StreamingResponse(BinaryIO):
+    _response: requests.Response
+    _buffer: bytes
+    _content: Iterator[any]
+
+    def __init__(self, response: requests.Response):
+        self._response = response
+        self._buffer = b''
+
+    def __enter__(self) -> BinaryIO:
+        self._content = self._response.iter_content()
+        return self
+
+    def close(self) -> None:
+        self._response.close()
+
+    def isatty(self) -> bool:
+        return False
+
+    def read(self, n: int = -1) -> AnyStr:
```

Comment:
Based on my reading of `AnyStr`, I think we should be a bit more narrow in terms of how we annotate this type. IMO `bytes` is more accurate. Looking at BinaryIO in typing.py:
```
class BinaryIO(IO[bytes]):
    ...
```
I think we can replace all instances of `AnyStr` with `bytes` in this class.